charts/cachemachine/templates/deployment.yaml

```diff
@@ -0,0 +1,72 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "cachemachine.fullname" . }}
+  labels:
+{{ include "cachemachine.labels" . | indent 4 }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: {{ include "cachemachine.name" . }}
+      app.kubernetes.io/instance: {{ .Release.Name }}
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: {{ include "cachemachine.name" . }}
+        app.kubernetes.io/instance: {{ .Release.Name }}
+    spec:
+    {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+    {{- end }}
+      serviceAccountName: {{ template "cachemachine.serviceAccountName" . }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      containers:
+        - name: {{ .Chart.Name }}
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          env:
+            {{- if .Values.docker_secret_name }}
+            - name: DOCKER_SECRET_NAME
+              value: {{ .Values.docker_secret_name }}
+            {{- end }}
+          ports:
+            - name: http
+              containerPort: 8080
+              protocol: TCP
+          livenessProbe:
+            httpGet:
+              path: /
+              port: http
+          readinessProbe:
+            httpGet:
+              path: /
+              port: http
+          resources:
+            {{- toYaml .Values.resources | nindent 12 }}
```

Comment:
I think the chart generator is a little aggro about adding too many things to `values.yaml`. I'd rather keep it minimal to make the chart easier to use, since then people don't have to think about whether they need to set all these things. Similar comment below about `affinity` and `nodeSelector` and whatnot.

Reply:
Yeah I'm using the chart generator.  I used to think as you did with the not parameterizing anything I don't need, although we've had a lot of things especially around labels and restricting services to particular nodes.  So I just left those things in.  I don't think it's so bad to have it be parameterized, but I'm not trying to add any unneeded parameterizing from my stuff.  One thing that has been nice about using the chart generator is that it sets up the labels and tolerations and resources nicely, and those seem to be at least somewhat useful.